### ISSUE TITLE
Error in VS Code when used in conjunction with eslint-plugin-prettier

### ISSUE THREAD

**Issue by TimonVS** [eslint/typescript-eslint-parser#431]
**What version of TypeScript are you using?**
2.6.2

**What version of `typescript-eslint-parser` are you using?**
11.0.0

**What code were you trying to parse?**

```ts
function a()
```

**What did you expect to happen?**

It should work nicely with eslint-plugin-prettier and not throw any errors.

**What happened?**

As soon as I type something like `function a()` or `const` I get the following error in VS Code:

![image](https://user-images.githubusercontent.com/876666/34768904-7369e93a-f5fc-11e7-8eb1-716110014c85.png)

Running eslint in the command line results in the following error:

```
Unexpected token, expected "{" (1:13)
> 1 | function a()
    |             ^
SyntaxError: Unexpected token, expected "{" (1:13)
> 1 | function a()
    |             ^
    at createError$1 (/Users/timon/Projects/Personal/eslint-typescript-parser-bug/node_modules/prettier/parser-babylon.js:1:112)
    at parse (/Users/timon/Projects/Personal/eslint-typescript-parser-bug/node_modules/prettier/parser-babylon.js:1:878)
    at Object.parse$1 [as parse] (/Users/timon/Projects/Personal/eslint-typescript-parser-bug/node_modules/prettier/index.js:5837:12)
    at formatWithCursor (/Users/timon/Projects/Personal/eslint-typescript-parser-bug/node_modules/prettier/index.js:27654:22)
    at format (/Users/timon/Projects/Personal/eslint-typescript-parser-bug/node_modules/prettier/index.js:27697:10)
    at Object.format (/Users/timon/Projects/Personal/eslint-typescript-parser-bug/node_modules/prettier/index.js:27936:12)
    at Program (/Users/timon/Projects/Personal/eslint-typescript-parser-bug/node_modules/eslint-plugin-prettier/eslint-plugin-prettier.js:374:45)
    at listeners.(anonymous function).forEach.listener (/Users/timon/Projects/Personal/eslint-typescript-parser-bug/node_modules/eslint/lib/util/safe-emitter.js:47:58)
    at Array.forEach (<anonymous>)
    at Object.emit (/Users/timon/Projects/Personal/eslint-typescript-parser-bug/node_modules/eslint/lib/util/safe-emitter.js:47:38)
    at NodeEventGenerator.applySelector (/Users/timon/Projects/Personal/eslint-typescript-parser-bug/node_modules/eslint/lib/util/node-event-generator.js:251:26)
    at NodeEventGenerator.applySelectors (/Users/timon/Projects/Personal/eslint-typescript-parser-bug/node_modules/eslint/lib/util/node-event-generator.js:280:22)
    at NodeEventGenerator.enterNode (/Users/timon/Projects/Personal/eslint-typescript-parser-bug/node_modules/eslint/lib/util/node-event-generator.js:294:14)
    at CodePathAnalyzer.enterNode (/Users/timon/Projects/Personal/eslint-typescript-parser-bug/node_modules/eslint/lib/code-path-analysis/code-path-analyzer.js:608:23)
    at Traverser.enter [as _enter] (/Users/timon/Projects/Personal/eslint-typescript-parser-bug/node_modules/eslint/lib/linter.js:998:32)
    at Traverser._traverse (/Users/timon/Projects/Personal/eslint-typescript-parser-bug/node_modules/eslint/lib/util/traverser.js:132:14)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

When I disable eslint-plugin-prettier, linting in VS Code works fine again.

I made a repository to easily reproduce the bug: https://github.com/TimonVS/eslint-typescript-parser-bug

**Comment by TimonVS** [comment]
Never mind! I apparently had to set the parser for Prettier.

```json
{
  "extends": ["eslint:recommended", "prettier"],
  "parserOptions": {
    "ecmaVersion": 8,
    "sourceType": "module"
  },
  "plugins": ["prettier"],
  "parser": "typescript-eslint-parser",
  "rules": {
    "semi": "off",
    "prettier/prettier": [
      "warn",
      {
        "parser": "typescript"
      }
    ]
  }
}
```
